### ISSUE TITLE
[Data Sharing] TextInput control missing role and label

### ISSUE THREAD

**Issue by svinkle** [CovidShield/mobile#52]
## Issue summary

The Code `TextInput` is missing a role description and a label. A role helps to describe what the control is, how to interact with it, and the expected outcome. A label give the control a name to convey its purpose. Without a role description and label, screen reader and voice dictation users will have great difficulty when filling out the form.

<details>
    <summary>Screenshots</summary>

![Screen Shot 2020-05-27 at 3 30 01 PM](https://user-images.githubusercontent.com/1392632/83064023-1ab1fa00-a02f-11ea-91ce-051ab43bc5b9.png)

</details>

#### TSX

```tsx
// CodeInput.tsx

<TextInput
  value={value}
  ref={inputRef}
  onChangeText={onChangeTrimmed}
  keyboardType="number-pad"
  …
/>
<TouchableWithoutFeedback onPress={giveFocus}>
  // …
</TouchableWithoutFeedback>

// …

const styles = StyleSheet.create({
  input: {
    height: 0,
    width: 0
  },
});
```

## Steps to reproduce

1. Load the app in the iOS simulator.
2. Open the Sheet view.
3. Activate the "Share code" control
4. Enable [VoiceOver screen reader](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) on macOS
5. Use the Virtual Cursor to navigate to the Code `TextInput` control

### Expected

- Control to be announce with a role description and label.

### Actual

- Role description and label are missing.

## Recommendation

1. To add a label, add the `accessibilityLabel` prop, setting its value to convey the control purpose.
2. To convey the role, adjust the `StyleSheet` of the `TextInput` to be transparent and positioned overtop of the custom output `Text` component.
3. To remove the visible caret, add the `caretHidden={true}` prop to the `TextInput`.
4. To remove the custom output from being focusable, add the `accessibilityElementsHidden={true}` (for iOS) and `importantForAccessibility="no-hide-descendants"` (for Android) props to the `TouchableWithoutFeedback` component.

#### TSX

```tsx
// CodeInput.tsx

<TextInput
  value={value}
  ref={inputRef}
  onChangeText={onChangeTrimmed}
  keyboardType="number-pad"
  …
  accessibilityLabel="Covid Shield Code"
  caretHidden={true}
/>
<TouchableWithoutFeedback 
  onPress={giveFocus}
  accessibilityElementsHidden={true}
  importantForAccessibility="no-hide-descendants"
>
  // …
</TouchableWithoutFeedback>

// …

const styles = StyleSheet.create({
  input: {
    borderWidth: 0,
    color: 'transparent',
    height: 40,
    left: 15,
    position: 'absolute',
    top: 0,
    width: '100%',
    zIndex: 1
  },
});
```

## Specifications

- **Component:** CovidShield [Data Sharing]
- **WCAG Principle:** Understandable
- **WCAG SC:** 3.3.2 Labels or Instructions
- **Severity:** High
- **Effort:** Medium
